Assets/Scripts/UI/Canvas/StateMachine/UIStateManager.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using Framework.PhoneCamera;
+using UnityEngine;
+
+using UI.Canvas.Introduction;
+using UI.Canvas.PhotoBookSystem;
+using UI.Canvas.PhotoTaking;
+
+namespace UI.Canvas.StateMachine
+{
+    public sealed class UIStateManager : MonoBehaviour
+    {
+        [Header("References")]
+        [SerializeField] private IntroductionPopup introductionPopup;
+        [SerializeField] private PhotoBook photoBook;
+        [SerializeField] private GameObject cameraPanel;
+        [SerializeField] private GameObject takingPicture;
+        
+        [Header("Current UI State")]
+        [SerializeField] private UIState currentState;
```

Comment:
```suggestion
        [SerializeField] private UIState currentState;

        [Header("References")]
        [SerializeField] private IntroductionPopup introductionPopup;
        [SerializeField] private PhotoBook photoBook;
        [SerializeField] private GameObject cameraPanel;
        [SerializeField] private GameObject takingPicture;
```
Het is nu "Current UI state: current state", dit vind ik niet echt mooi, dus verplaats het naar boven zonder header. Waardoor alleen de refrences een header hebben.

OF

Bedenk een betere naam voor de 2de header.